lua/fff/rust/file_picker.rs

```diff
@@ -244,12 +247,26 @@ impl FilePicker {
         let total_files = sync_data.files.len();
 
         // small queries with a large number of results can match absolutely everything
-        let max_typos = (query.len() as u16 / 4).clamp(2, 6);
+        let max_typos = if query.len() <= 4 {
+            0
+        } else {
+            (query.len() as u16 / 5).clamp(1, 3)
+        };
```

Comment:
I found the previous `max_typos` value to be too permissive, which was problematic when scoring against proximity. I primarily work within a large TypeScript monorepo, and when typing `Page` it didn't filter out any items.

This change should hopefully be a bit more adaptive. I wouldn't be surprised if we would need to tweak this though.